tests/unit/alp_cholesky.cpp

```diff
@@ -22,71 +22,169 @@
 #include <alp.hpp>
 #include <alp/algorithms/cholesky.hpp>
 #include <alp/utils/parser/MatrixFileReader.hpp>
+#include "../utils/print_alp_containers.hpp"
 
 using namespace alp;
 
-void alp_program( const size_t & unit, alp::RC & rc ) {
-	rc = SUCCESS;
+typedef double ScalarType;
+constexpr ScalarType tol = 1.e-12;
+
+struct inpdata {
+	std::string fname;
+};
+
+//** check the solution by calculating the Frobenius norm of (H-LL^T) **//
+template< typename T, typename RingType, typename ZeroType >
+alp::RC check_cholesky_solution(
+	const  alp::Matrix< T, structures::Symmetric, Dense > &H,
+	alp::Matrix< T, structures::UpperTriangular, Dense > &L,
+	const ZeroType &zero_scalar,
+	const RingType &ring
+) {
+	alp::RC rc = SUCCESS;
+	const size_t N = nrows( H );
+	alp::Matrix< T, alp::structures::Symmetric > LLT( N, N );
+	alp::set( LLT, zero_scalar );
+	auto LT = alp::get_view< alp::view::transpose >( L );
+#ifdef DEBUG
+	print_matrix( " << LLT >> ", LLT );
+	print_matrix( " << LT >>  ", LT );
+#endif
+	alp::mxm( LLT, LT, L, ring );
+#ifdef DEBUG
+	print_matrix( " << LLT >> ", LLT );
+#endif
+
+	alp::Matrix< T, alp::structures::Symmetric > TMP( N, N );
+	alp::set( TMP, zero_scalar );
```

Comment:
I suggest using a more informative name for `TMP`, e.g. `residual` or something along these lines. Also I suggest moving the declaration below, closer to the first use.